codegen/module.go

```diff
@@ -868,6 +870,15 @@ func (system *ModuleSystem) GenerateBuild(
 		return nil, err
 	}
 
+	serializedModules, err := json.Marshal(resolvedModules)
```

Comment:
Instead of JSON, maybe Gob?

Reply:
neat +1

Reply:
On second thought, maybe we should leave it open for bash scripting using `jq`; it's not being transferred over the network yet.